parallel/runner.go

```diff
@@ -145,3 +150,34 @@ func (r *runner) Cancel() {
 func (r *runner) Errors() map[int]error {
 	return r.errors
 }
+
+// Define the work as done when all consumers are idle for idleThresholdSeconds.
```

Comment:
I suggest adding after the above - 
The function will wait until all consumers are idle.